src/RAMI.cpp

```diff
@@ -654,13 +659,13 @@ void RAMI::evalOp(const RamOperation& op, const InterpreterContext& args) {
             // run nested part - using base class visitor
             if (aggregate.getFunction() == souffle::MAX || aggregate.getFunction() == souffle::MIN) {
                 if (res == (aggregate.getFunction() == souffle::MAX ? MIN_RAM_DOMAIN : MAX_RAM_DOMAIN)) {
-                    return;
+                    return true;
```

Comment:
This nested if seems unnecessarily confusing. While you're modifying it, maybe change it to
```
if (agg.func == max && res == MIN_RAM_DOMAIN) {
  return true;
}
if (agg.func == min ...
```